packages/tofu/src/inspectSesCompat.js

```diff
@@ -37,9 +66,15 @@ function inspectSesCompat(ast) {
     }
   })
   // check for mutations to named intrinsics
-  const sesNamedIntrinsics = Reflect.ownKeys(sesAllowlist).filter(
-    (k) => k in global && typeof sesAllowlist[k] === 'object'
-  )
+  const sesNamedIntrinsics = Reflect.ownKeys(sesAllowlist)
+    .filter(
+      (k) =>
+        k in global &&
+        // @ts-expect-error - filter loses specificity
```

Comment:
alternatively, I could make a type assertion, e.g., `/** @type {keyof sesAllowList} */(k)` or `if (k in sesAllowList)`

Reply:
The reason this happens is because `Reflect.ownKeys()` just returns `Array<string|symbol>` instead of the discrete list of keys in `sesAllowList`. So all we know is we have some `string`s and `symbol`s and do not know that they have anything to do with `sesAllowList`.

Reply:
To that end, I'm curious why we need to use `Reflect.ownKeys()` here?  @kumavis